src/main/java/graphql/GraphQL.java

```diff
@@ -150,23 +152,71 @@ public ExecutionResult execute(String requestString) {
     }
 
     public ExecutionResult execute(String requestString, Object context) {
-        return execute(requestString, context, Collections.<String, Object>emptyMap());
+        return execute(requestString, context, Collections.emptyMap());
     }
 
     public ExecutionResult execute(String requestString, String operationName, Object context) {
-        return execute(requestString, operationName, context, Collections.<String, Object>emptyMap());
+        return execute(requestString, operationName, context, Collections.emptyMap());
     }
 
     public ExecutionResult execute(String requestString, Object context, Map<String, Object> arguments) {
         return execute(requestString, null, context, arguments);
     }
 
+
     public ExecutionResult execute(String requestString, String operationName, Object context, Map<String, Object> arguments) {
-        InstrumentationContext<ExecutionResult> executionCtx = instrumentation.beginExecution(new ExecutionParameters(requestString, operationName, context, arguments));
+        return executeAsynch(requestString, operationName, context, arguments).join();
+    }
+
+    /**
+     * This will return a promise (aka {@link CompletableFuture}) to provide a {@link ExecutionResult}
+     * which is the result of executing the provided query.
+     *
+     * @param requestString the query to be executed
+     * @param operationName the name of the operation
+     * @param context       a context object for the query
+     * @param arguments     the arguments for the query
+     *
+     * @return a promise to an execution result
+     */
 
+    public CompletableFuture<ExecutionResult> executeAsynch(String requestString, String operationName, Object context, Map<String, Object> arguments) {
```

Comment:
Since I want to be able to .join() in a variety of places I think we should leave this as CompleteableFuture